### PR TITLE
Release 3.0.1

### PR DESCRIPTION
**Pull request by panjan**

### CODE REVIEW

File: lib/gooddata/connection.rb [hunk 1]
```diff
@@ -118,7 +118,7 @@ def sso_url(login, provider, url = GoodData::Rest::Connection::DEFAULT_URL, opts
     # @param [String] login Email address used for logging into gooddata
     # @param [String] provider Name of SSO provider
     # @return [GoodData::Rest::Client] Instance of REST client
-    def connect_sso(login, provider, url, opts)
+    def connect_sso(login, provider, url = GoodData::Rest::Connection::DEFAULT_URL, opts = {})
```
Comment (yenkins) [inline lib/gooddata/connection.rb]: Has 4 parameters (LongParameterList)

File: lib/gooddata/connection.rb [hunk 2]
```diff
@@ -66,7 +66,7 @@ def with_connection(options = nil, second_options = nil, third_options = {}, &bl
     # @param [Hash] opts Additional options
     # @option opts [Fixnum] :validity Validity in seconds from 'now'
     # @return [String] URL which can be used for SSO logging in
-    def sso_url(login, provider, url = GoodData::Rest::Connection::DEFAULT_URL, opts = DEFAULT_SSO_OPTIONS)
+    def sso_url(login, provider, url, opts = DEFAULT_SSO_OPTIONS)
```
Comment (yenkins) [inline lib/gooddata/connection.rb]: Has 4 parameters (LongParameterList)

Comment (yenkins) [inline lib/gooddata/connection.rb]: Has approx 22 statements (TooManyStatements)

File: lib/gooddata/lcm/helpers/check_helper.rb [hunk 3]
```diff
@@ -16,18 +16,21 @@ def check_params(specification, params)
             type = specification[param_name][:type]
             if value.nil? || (value.is_a?(String) && value.empty?)
               if specification[param_name][:opts][:default]
-                params[param_name] = specification[param_name][:opts][:default]
+                if specification.select { |x| specification[x][:opts][:replacement] == param_name }.first.nil?
+                  params[param_name] = specification[param_name][:opts][:default]
+                else
+                  GoodData.logger.warn "WARNING: Default value for parameter '#{param_name}' was not filled because deprecated parameter is used instead."
```
Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'gooddata.logger' 2 times (DuplicateMethodCall)

File: lib/gooddata/lcm/helpers/check_helper.rb [hunk 4]
```diff
@@ -16,18 +16,21 @@ def check_params(specification, params)
             type = specification[param_name][:type]
             if value.nil? || (value.is_a?(String) && value.empty?)
               if specification[param_name][:opts][:default]
-                params[param_name] = specification[param_name][:opts][:default]
+                if specification.select { |x| specification[x][:opts][:replacement] == param_name }.first.nil?
+                  params[param_name] = specification[param_name][:opts][:default]
```
Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'specification[param_name]' 6 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'specification[param_name][:opts]' 5 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'specification[param_name][:opts][:default]' 2 times (DuplicateMethodCall)

File: lib/gooddata/lcm/helpers/check_helper.rb [hunk 5]
```diff
@@ -16,18 +16,21 @@ def check_params(specification, params)
             type = specification[param_name][:type]
             if value.nil? || (value.is_a?(String) && value.empty?)
               if specification[param_name][:opts][:default]
-                params[param_name] = specification[param_name][:opts][:default]
+                if specification.select { |x| specification[x][:opts][:replacement] == param_name }.first.nil?
```
Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Refers to 'specification' more than self (maybe move it to another class?) (FeatureEnvy)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Contains iterators nested 2 deep (NestedIterators)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Performs a nil-check (NilCheck)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Has the variable name 'x' (UncommunicativeVariableName)

File: lib/gooddata/lcm/lcm2.rb [hunk 6]
```diff
@@ -365,6 +359,17 @@ def perform(mode, params = {})
         }
       end
 
+      def run_action(action, params)
+        GoodData.logger.info("Running #{action.name} action ...")
+        params.clear_filters
+        # Check if all required parameters were passed
+        BaseAction.check_params(action.const_get('PARAMS'), params)
```
Comment (yenkins) [inline lib/gooddata/lcm/lcm2.rb]: Calls 'action.const_get('params')' 2 times (DuplicateMethodCall)

File: lib/gooddata/lcm/lcm2.rb [hunk 7]
```diff
@@ -365,6 +359,17 @@ def perform(mode, params = {})
         }
       end
 
+      def run_action(action, params)
+        GoodData.logger.info("Running #{action.name} action ...")
+        params.clear_filters
```
Comment (yenkins) [inline lib/gooddata/lcm/lcm2.rb]: Calls 'params.clear_filters' 2 times (DuplicateMethodCall)

File: lib/gooddata/lcm/lcm2.rb [hunk 8]
```diff
@@ -365,6 +359,17 @@ def perform(mode, params = {})
         }
       end
 
+      def run_action(action, params)
```
Comment (yenkins) [inline lib/gooddata/lcm/lcm2.rb]: Has approx 7 statements (TooManyStatements)

File: lib/gooddata/models/user_filters/user_filter_builder.rb [hunk 9]
```diff
@@ -385,7 +385,14 @@ def self.execute_mufs(user_filters, options = {})
       else
         GoodData.logger.warn("Data permissions computed: #{to_create.count} to create and #{to_delete.count} to delete")
       end
-      return { created: to_create, deleted: to_delete } if dry_run
+
+      if dry_run
+        create_results = to_create.map { |x| { status: 'dry_run', user: x.first, type: 'create' } }
```
Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Calls 'x.first' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Has the variable name 'x' (UncommunicativeVariableName)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/76/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/887/ : SUCCESS in 43s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1155/ : SUCCESS in 36s

**Comment by kubamahnert** [comment]
ok to merge

**Comment by kubamahnert** [comment]
Thanks @panjan 👌👌

**Comment by gdgate** [comment]
Merge Failed.

This change or one of its cross-repo dependencies was unable to be automatically merged with the current state of its repository. Please rebase the change and upload a new patchset.

File: lib/gooddata/lcm/helpers/check_helper.rb [hunk 10]
```diff
@@ -16,7 +16,11 @@ def check_params(specification, params)
...
+                  params[param_name] = specification[param_name][:opts][:default]
+                else
+                  GoodData.logger.warn "WARNING: Default value for parameter '#{param_name}' was not filled because deprecated parameter is used instead."
```
Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'gooddata.logger' 5 times (DuplicateMethodCall)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/92/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/909/ : SUCCESS in 54s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1177/ : SUCCESS in 44s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/24/ : SUCCESS in 16s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/910/ : SUCCESS in 45s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1178/ : SUCCESS in 40s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1358/ : SUCCESS in 57m 48s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/1121/ : SUCCESS in 34m 18s

**Pull request merged**